tests/fgcmcalTestBase.py

```diff
@@ -551,6 +549,94 @@ def _testFgcmCalibrateTract(self, visits, tract,
         self.assertFalse(butler.datasetExists('fgcmStarIndices'))
         self.assertFalse(butler.datasetExists('fgcmReferenceStars'))
 
+    def _compareBuildStars(self, butler1, butler2):
+        """
+        Compare the full set of BuildStars outputs with files from two
+        repos.
+
+        Parameters
+        ----------
+        butler1 : `lsst.daf.persistence.Butler`
+        butler2 : `lsst.daf.persistence.Butler`
+
+        Raises
+        ------
+        Exceptions on test failures.
+        """
+        import numpy.testing as testing
+        import esutil
+
+        # Check the visit catalogs are identical
+        visitCat1 = butler1.get('fgcmVisitCatalog').asAstropy()
+        visitCat2 = butler2.get('fgcmVisitCatalog').asAstropy()
+
+        for col in visitCat1.columns:
+            if isinstance(visitCat1[col][0], str):
+                testing.assert_array_equal(visitCat1[col], visitCat2[col])
+            else:
+                testing.assert_array_almost_equal(visitCat1[col], visitCat2[col])
+
+        # Check that the observation catalogs have the same length
+        starObs1 = butler1.get('fgcmStarObservations')
+        starObs2 = butler2.get('fgcmStarObservations')
+        self.assertEqual(len(starObs1), len(starObs2))
+
+        # Check that the number of stars is the same and all match
+        starIds1 = butler1.get('fgcmStarIds')
+        starIds2 = butler2.get('fgcmStarIds')
+        self.assertEqual(len(starIds1), len(starIds2))
+
+        # htm = esutil.htm.HTM(11)
+        matcher = esutil.htm.Matcher(11, starIds1['ra'], starIds1['dec'])
+        matches = matcher.match(starIds2['ra'], starIds2['dec'], 1./3600., maxmatch=1)
+
+        self.assertEqual(len(matches[0]), len(starIds1))
+
+        # Check that the number of observations of each star is the same
+        testing.assert_array_equal(starIds1['nObs'][matches[1]],
+                                   starIds2['nObs'][matches[0]])
```

Comment:
I think I would remove the newlines from this block, since these all are part of one logical test. Or possibly make this a separate `_test_fgcmStarIds_output()` method, or something (similarly for other blocks), though it may not be worth it.